### PR TITLE
chore(dtslint): comment helper functions

### PR DESCRIPTION
**Pull request by cartant**
**Description:**

This PR adds comments explaining the purpose of the `a` helper functions that are used in the dtslint tests for `Observable.prototype.pipe` and the static `pipe` function.

Without the helper functions the tests would involve much more code, but what the functions actually do isn't immediately obvious, as they depend upon the inference and compatibility/incompatibility of string literal types.

**Related issue (if exists):** None

### CODE REVIEW

File: spec-dtslint/Observable-spec.ts [hunk 1]
```diff
@@ -1,6 +1,22 @@
 import { Observable, of, OperatorFunction } from 'rxjs';
 import { mapTo } from 'rxjs/operators';
 
+// The `a` (for argument) function is used to keep the tests uncluttered.
```
Comment (benlesh) [inline spec-dtslint/Observable-spec.ts]: If you do a `/** Info here */` style doc-comment, IDEs like VS Code will pick them up. :)

Comment (cartant) [reply spec-dtslint/Observable-spec.ts]: Done.

**Comment by benlesh** [comment]
@cartant looks like Travis is failing on this one because of a bad type in the comments.

**Comment by cartant** [comment]
@benlesh This should be fixed. I've disabled some picky TSLint rules that defaulted to being enabled in the dtslint configuration.

**Pull request merged**